Rome Incentive Structure (#41)
Joystream/communications: blog/rome-incentives/README.md

# Post

#### Title

The Rome Incentive Structure

#### Purpose

Inform our community about Rome's incentive structure.

#### url

blog.joystream.org/rome-incentives/

#### Cover

<p align="center"><img src="rome-incentives.png"></p>

#### Lead

Rome offers a new incentive structure for our testnet participants.

#### Body

> As with Sparta, Athens and Acropolis we plan to pay users in monero (XMR) to participate on our upcoming testnet release, codenamed Rome. Our fifth testnet ambitiously introduces two new paid roles which we are extremely excited about!

Reviewer: - Perhaps add a link to previous post, and announce the release date for this testnet
- Add that guides will be up Sunday
- Finally, add that unlike previous testnets, this release will have *dynamic* incentives, in order to
   - Allocate resources
   - Ensure good uptime/response/etc.
   - Prepare for next testnet, that will shift the responsibility of setting rewards to the council

Reviewer: (Doesn't have to be a bullet point list)

Executive Summary
-----------------

- `Content Creators` receive up to $0.60 per qualifying item of content uploaded\
- `Content Curators` earn either $30 or $60 per week, depending on workload\
- `Storage Providers` will share $300 per week, plus $0.025/GB per week\
- `Validators` are competing for their share of $50 per week\
- `Council Members` are paid $10 for getting elected, with a $10 bonus\
- `Bug Reporters` can continue to earn monero for reporting and/or fixing bugs

Who, Why, What and How?
-----------------------

> As with previous incentivized testnets, anyone who wants to can participate on Rome and earn monero, but there is some extra help and guidance for those that ask for it in our [Telegram group](https://t.me/JoyStreamOfficial).

Reviewer: Add that the faucet will only include smaller payouts, whereas tlgrm, twtr and mailing list subs can ask for tokens...

The philosophy behind why was discussed in detail in a [previous post](https://blog.joystream.org/pay-for-play/). In short, we are paying to prepare active participants for an autonomous platform and test the incentive structure with "real money".

What you can do to qualify for getting paid on Rome is described below. More information, guides on how to get setup and extra support can be found in our helpdesk repo on GitHub.

Paid Roles
----------

- Content Creator
- Content Curator
- Storage Provider
- Validator
- Council Member
- Bug Reporter

If you find one or more of these roles interesting, or want to learn about future roles on the network, we encourage you to visit the new [roles](https://joystream.org/roles) section on our website. Here you can browse through them and get a sense of what they will mean to the network, along with responsibilities and requirements.

You can also sign up for role specific newsletters that will keep you updated on development and incentives!

Content Creators
----------------

![](Content%20Creators.png)

> <i>Compete for $350 per week as a content creator</i>

Reviewer: I think we have to revise the thinking here. Something like paying for first `n` videos a week, but always payout for original content?

For the Rome testnet, we plan to pay a flat-rate Monero reward for every piece of content uploaded by `Content Creators` (with a different rate for each content type). We will also pay a $1.50 bonus on top of this rate for the 20 best pieces of content that are uploaded every week to encourage higher quality uploads. The maximum weekly incentive pool of $350 will be divided between these two categories of incentive.

Prospective `Content Creators` should also be aware that we will only pay the item reward for content with complete metadata and which follows the set-out requirements for each content type (to be released at a later date).

The `Content Curators` (see below) will be incentivized to find content with incorrect metadata and will likely remove it from the content directory. In these cases, content creators will not be paid. Multiple spam uploads which represent a burden to moderate for the `Content Curators` may even be penalized and result in deductions on payouts due for qualifying content uploads on your content creator profile.

> Uploading illegal or copyrighted content will result in a disqualification from payouts. We will provide `Content Creators` with a list of sources for safe and non-copyrighted material which can be uploaded as part of this testnet and will qualify for the payments discussed above and laid out in the table below.

Reviewer: ...and takedown of content, potentially slashing of funds, and deleting channel.

### Rewards based on content type

| CONTENT TYPE | MAXIMUM PAYMENT PER ITEM | MAXIMUM WEEKLY POOL |
| --- | --- | --- |
| Short Video (Standard) | $0.35 per video* | $70 |
| Long Video (Standard) | $0.50 per video* | $100 |

> | Album Item (Standard)** | $0.50 per track* | $100 |

Reviewer: Music not yet available...

In general, I think this needs a little more thinking.

Basically, the payment per item is constant until the number of items in
category multiplied by maximum payment per item exceeds the maximum weekly
pool. We have chosen this reward structure in order to incentivize users to
upload a variety of different content types on Rome.

Payouts for this role will be made every Monday at ~11:00GMT.

Content Curators
----------------

![](Content%20Curators.png)

<i>Earn your share of $375 per week</i>

`Content Curators` will one day be essential for ensuring that the petabytes of media items uploaded to Joystream are formatted correctly and comprehensively monitored and moderated. Our upcoming testnet allows this content monitoring to take place by giving users who are selected for the role administrative access to the Joystream content directory to make changes where necessary.

We will have five slots available for `Content Curators` when Rome is launched. They will need to apply to the `Content Curator Lead` (a `sudo` role operated by [Jsgenesis](https://jsgenesis.com/)) and will have their application assessed by this lead before being added as a curator for a default term of one week.

> If successful in their application, curators will be rewarded with a flat reward of `$60` per week, resulting in a maximum pool of `$300` if all five content curators are signed up. If there are less than 100 pieces of content uploaded in the week in which a curator is operating, the individual payout will be reduced to `$30` for that week, resulting in a total pool of `$150`.

Reviewer: Maybe make this amount smaller?

- It seems like it could start lower, but include pay per action?

> There will also be two bonuses of `$50` and `$25` per week for the two most active curators respectively in terms of monitoring the items in the content directory for compliance with our metadata, category and content rules. These bonuses will be awarded by the `Content Curator Lead`. In total, during a typical week, `Content Curators` will be competing for their share of `$375` in XMR.

Reviewer: I like this. Perhaps we should shift towards higher bonus, lower base?

Also add that inactive and malicious curators will face slashing and booting...

Payouts will be made every Monday at ~11:00GMT.

Storage Providers
-----------------

![](Storage%20Providers.png)

> <i>Compete for $300 per week, plus bonus for data</i>

Reviewer: I think this should be skewed heavily towards data and bandwidth cost. This certainly needs to be dynamic position...

Unlike `Validators` that can come and go without too much friction (at least for now), a new `Storage Provider` replicates the entire content directory. As a consequence, the platform needs some stability for this role to avoid a poor user experience, or worse, loss of data. After migrating to the Rome testnet, the users that currently hold the role will get "kicked" out, and will have to sign up again if they still want to keep their role.

> Up to 10 `Storage Providers` that keep a full copy and provides continuous service for at least 24h will compete for `$300 per week`. In addition,  you will earn `$0.025/GB/week` (of data in the content directory) calculated on an average basis. You will also need a domain to point to your node, so unless you already own one (that you don't mind using), you will need to buy one.

Reviewer: Let's start out with 8, but add that we will be far heavier on the hammer this time. Perhaps the curators could also be responsible for reporting on poor service?

Validators
----------

![](Validators.png)

<i>Compete for $50 per week</i>

In proof of stake systems, block producers, or `Validators`, are typically paid a fixed amount for each block produced. For Rome, the payout pool will remain the same at `$50` per week shared between all `Validators` on the network.

```
blocktime = 6
weekly_reward = 5000
seconds_in_week = 60*60*24*7

blockreward = (weekly_reward * blocktime)/seconds_in_week
print(blockreward)

----

0.05

```

The number - 0.05 cents per block - seems a bit underwhelming, but validation requires little effort for the user after setup, and blocks come in every 6secs. With armv7 binaries or low-end VPS nodes, it should be cheap to run! A detailed guide to setting up the `Validator` node and settings can be found [here](https://github.com/Joystream/helpdesk/tree/master/roles/validators).

After migrating to the Rome testnet, the users that currently hold the role will get "kicked" out, and will have to sign up again if they still want to keep their role.

Payouts will be made every Monday at ~11:00GMT.

> Council Members

Reviewer: This seems good for now. Won't be much work until next TN

# Post

#### Title

The Rome Incentive Structure

#### Purpose

Inform our community about Rome's incentive structure.

#### url

blog.joystream.org/rome-incentives/

#### Cover

<p align="center"><img src="rome-incentives.png"></p>

#### Lead

Rome offers a new incentive structure for our testnet participants.

#### Body

As with Sparta, Athens and Acropolis we plan to pay users in monero (XMR) to participate on our upcoming testnet release, codenamed Rome. Our fifth testnet ambitiously introduces two new paid roles which we are extremely excited about!

Rome will differ slightly from Acropolis in that it will have dynamic incentives in order to allocate resources more efficiently, ensure good uptime for validators and storage providers, promtote responsiveness and activity by content creators and curators and to help prepare for our next testnet, which will shift the responsibility of setting rewards from Jsgenesis to the council. This means that all of the compensation structures listed below are subject to review at any time. Once Rome is launched we will create a forum post where the payouts for each role will be reviewed every week.

> Rome is set to be launched on Monday 16th March. Read more about the procedure for the release [here](https://blog.joystream.org/from-acropolis-to-rome/). Updated guides for testnet roles will be released on Sunday 15th March. Prospective participants should familiarize themselves with these guides before launch for the best chance of winning a spot in one of these competitive roles.

Reviewer: I think we should include a note that we may be delayed due to illness...

> Executive Summary

Reviewer: Add that we will revise these numbers as we see fit, but that these will be the starting numbers, and we will never adjust *down* in the middle of a week!

-----------------

- `Content Creators` receive up to $0.50 per qualifying item of content uploaded plus bonuses
- `Content Curators` earn either $15 or $30 per week, depending on workload
- `Storage Providers` will share $120 per week, plus $0.03/GB per week

> - `Validators` are competing for their share of $50 per week

Reviewer: I think this number should go up to $100!

What you can do to qualify for getting paid on Rome is described below. More information, guides on how to get setup and extra support can be found in our helpdesk repo on GitHub.

Participants should also note that some important changes to the faucet will take place upon the launch of Rome. To prevent Sybil attacks, tokens will be made available primarily through our communication channels (including Telegram, Twitter and our mailing list). The faucet will continue to operate, but will offer considerably lower payouts.

Paid Roles
----------

- Content Creator
- Content Curator
- Storage Provider
- Validator
- Council Member
- Bug Reporter

If you find one or more of these roles interesting, or want to learn about future roles on the network, we encourage you to visit the new [roles](https://joystream.org/roles) section on our website. Here you can browse through them and get a sense of what they will mean to the network, along with responsibilities and requirements.

You can also sign up for role specific newsletters that will keep you updated on development and incentives!

> Content Creators

Reviewer: This seems a little overwhelming.

Maybe it can be a little more concise? If you don't mind, I'll make a suggestion?

For example, if more than 200 `short videos` are uploaded in a week the reward per
short video item will be reduced for that week. In the case of 400 `short videos` being
uploaded, the reward will be halved to $0.175 per file ($70 ÷ 400). For 800
`short videos` in a week the reward would be $0.0875 and so on.

Basically, the payment per item is constant until the number of items in
category multiplied by maximum payment per item exceeds the maximum weekly
pool. We have chosen this reward structure in order to incentivize users to
upload a variety of different content types on Rome.

Payouts for this role will be made every Monday at ~11:00GMT.

Content Curators
----------------

![](Content%20Curators.png)

<i>Earn your share of $375 per week</i>

`Content Curators` will one day be essential for ensuring that the petabytes of media items uploaded to Joystream are formatted correctly and comprehensively monitored and moderated. Our upcoming testnet allows this content monitoring to take place by giving users who are selected for the role administrative access to the Joystream content directory to make changes where necessary.

> We will have five slots available for `Content Curators` when Rome is launched. They will need to apply to the `Content Curator Lead` (a `sudo` role operated by [Jsgenesis](https://jsgenesis.com/)) and will have their application assessed by this lead before being added as a curator for a default term of one week.

Reviewer: We will open with 5 slots, but this can be changed depending on demand and activity.

The group lead will initially be occupied by Jsgenesis, but as the network matures, the role will be delegated to a community member.

If successful in their application, curators will be rewarded with a flat reward of `$30` per week, resulting in a maximum pool of `$150` if all five content curators are signed up. If there are less than 100 pieces of content uploaded in the week in which a curator is operating, the individual payout will be reduced to `$15` for that week, resulting in a total pool of `$75`.

There will also be two bonuses of `$20` and `$10` per week for the two most active curators respectively in terms of monitoring the items in the content directory for compliance with our metadata, category and content rules. These bonuses will be awarded by the `Content Curator Lead`. In total, during a typical week, `Content Curators` will be competing for their share of `$375` in XMR.

> Inactive malicious curators will be slashed and booted by the `Content Curator Lead`.

Reviewer: ```suggestion
Inactive and/or malicious curators will be booted, and potentially slashed by the `Content Curator Lead`.
```

Reviewer: Did you not agree with this one?

Up to eight `Storage Providers` that keep a full copy of the content directory and provide continuous service for at least 24h will compete for `$120 per week`. In addition,  you will earn `$0.03/GB/week` (of data in the content directory) calculated on an average basis. You will also need a domain to point to your node, so unless you already own one (that you don't mind using), you will need to buy one.

We will be closely monitoring `Storage Providers` for Rome (more so than we did for Acropolis) so make sure your node is performing optimally to avoid being booted. `Storage Providers` should also be aware that the variable part of the compensation structure is likely to be reviewed after launch (up or down) to make sure that providers are fairly compensated for their efforts.

Payouts will be made every Monday at ~11:00GMT.

> A detailed guide to setting up the node [can be found here](https://github.com/Joystream/helpdesk/tree/master/roles/storage-providers).

Reviewer: Remove this for now. Will be addressed later!

Validators
----------

![](Validators.png)

> <i>Compete for $50 per week</i>

Reviewer: Number go up :)

```
blocktime = 6
weekly_reward = 5000
seconds_in_week = 60*60*24*7

blockreward = (weekly_reward * blocktime)/seconds_in_week
print(blockreward)

----

0.05

```

The number - 0.05 cents per block - seems a bit underwhelming, but validation requires little effort for the user after setup, and blocks come in every 6secs. With armv7 binaries or low-end VPS nodes, it should be cheap to run! A detailed guide to setting up the `Validator` node and settings can be found [here](https://github.com/Joystream/helpdesk/tree/master/roles/validators).

After migrating to the Rome testnet, the users that currently hold the role will get "kicked" out, and will have to sign up again if they still want to keep their role.

Payouts will be made every Monday at ~11:00GMT.

Council Members
---------------

![](Council%20Members.png)

<i>Earn $10 per election cycle</i>

Council Members are elected by the stakeholders in the system to act in the interest of their constituency. Somewhat simplified, the Council will allocate the platform's resources, and hire executive personnel to run the day to day. We are looking at how to best incentivize them to act in the platform's long-term interest.

> If during you term a proposal to upgrade the runtime is submitted by the `sudo` key, `5CJzTaCp5fuqG7NdJQ6oUCwdmFHKichew8w4RZ3zFHM8qSe6` anyone that votes `yes` will receive a bonus worth $10. A detailed explanation the election cycle and responsibilities can be found [here](https://github.com/Joystream/helpdesk/tree/master/roles/council-members).

Reviewer: Forgot to add that the `sudo` key will be changed for Rome.

![](Content%20Creators.png)

> <i>Compete for $100 per week as a content creator</i>

Reviewer: ```suggestion
<i>Compete for up to $250 per week as a content creator</i>
```

When the Joystream mainnet is live, `Content Creators` will be responsible for creating and uploading the enormous variety of different content types and genres which we believe will allow Joystream to grow into a successful decentralized media platform.

> For the Rome testnet, we will pay a flat-rate Monero reward for every piece of content uploaded by `Content Creators` up to a weekly limit. We will also pay a $1 bonus on top of this rate for each of the 25 best pieces of content that are uploaded every week to encourage higher quality uploads. A separate $1 bonus will be paid for the first 25 items of "original"* content every week. The maximum weekly incentive pool of $200 will be divided between these two categories of incentive.

Reviewer: ```suggestion
For the Rome testnet, we will pay a flat-rate Monero reward for every piece of content uploaded by `Content Creators` up to a weekly limit. We will also pay a $5 bonus on top of this rate for each of the 10 best pieces of content that are uploaded every week to encourage higher quality uploads. 

A separate $10 bonus will be paid for up to 10 items of quality "original"* content every week. The maximum weekly incentive pool of $250 will be divided between these two categories of incentive.
```

* Original here means creative work that you created yourself and own exclusive the rights to.

Prospective `Content Creators` should also be aware that we will only pay the item reward for content with complete metadata and which follows the set-out requirements for each content type (to be released at a later date).

Uploading illegal or copyrighted content will result in a disqualification from payouts. It will also result in the takedown of content, potentially slashing of funds, and the deletion of your channel. Multiple spam uploads which represent a burden to moderate for the `Content Curators` may even be penalized and result in deductions on payouts due for qualifying content uploads on your content creator profile.

We will provide `Content Creators` with a list of sources for safe and non-copyrighted material which can be uploaded as part of this testnet and will qualify for the payments discussed above and laid out in the table below.

### Rewards based on content type

| CONTENT TYPE | MAXIMUM PAYMENT PER ITEM | MAXIMUM QUALIFYING UPLOADS |  MAXIMUM PAYOUT POOL |
| --- | --- | --- |  --- |

> | Video (Standard) | $0.50 per video | 100 | $50 |

Reviewer: ```suggestion
| Video (Standard) | $0.50 per video | 200 | $100 |
```

> | Best Content (Bonus) | $1 per video | 25 | $25 |

Reviewer: ```suggestion
| Best Content (Bonus) | $5 per video | 10 | $50 |
```

> | Original Content (Bonus) | $1 per video | 25 | $25 |

Reviewer: ```suggestion
| Original Content (Bonus) | $10 per video | 10 | $100 |
```

> Payments will only be made for the first `X` items per week out of the quota which are uploaded other than for `best` content, for which the whole week's uploads will be reviewed.

Reviewer: ```suggestion
Payments will only be made for the first `X` items per week out of the quota which are uploaded other than for `best` and `original` content, for which the whole week's uploads will be reviewed.
```

`Content Curators` will one day be essential for ensuring that the petabytes of media items uploaded to Joystream are formatted correctly and comprehensively monitored and moderated. Our upcoming testnet allows this content monitoring to take place by giving users who are selected for the role administrative access to the Joystream content directory to make changes where necessary.

> We will have five slots available for `Content Curators` when Rome is launched, but this may be modified depending on demand and activity. They will need to apply to the `Content Curator Lead` (a `sudo` role operated by [Jsgenesis](https://jsgenesis.com/)) and will have their application assessed by this lead before being added as a curator for a default term of one week. As the network matures, the `Content Curator Lead` role will be delegated to a community member.

Reviewer: ```suggestion
We will have five slots available for `Content Curators` when Rome is launched, but this may be modified depending on demand and activity. They will need to apply to the `Content Curator Lead` (a working group lead role operated by [Jsgenesis](https://jsgenesis.com/)) and will have their application assessed by this lead before being added as a curator for a default term of one week. As the network matures, the `Content Curator Lead` role will be delegated to a community member.
```

> If successful in their application, curators will be rewarded with a flat reward of `$30` per week, resulting in a maximum pool of `$150` if all five content curators are signed up. If there are less than 100 pieces of content uploaded in the week in which a curator is operating, the individual payout will be reduced to `$15` for that week, resulting in a total pool of `$75`.

Reviewer: ```suggestion
If successful in their application, curators will be rewarded with a flat reward of `$30` per week, resulting in a maximum pool of `$150` if all five content curators are signed up, and stays in the role the entire week.
```
